exporter/opentelemetry-exporter-prometheus/tests/test_mapping.py

```diff
@@ -0,0 +1,113 @@
+# Copyright The OpenTelemetry Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from unittest import TestCase
+
+from opentelemetry.exporter.prometheus._mapping import (
+    map_unit,
+    sanitize_attribute,
+    sanitize_full_name,
+)
+
+
+class TestMapping(TestCase):
+    def testsanitize_full_name(self):
+        self.assertEqual(
+            sanitize_full_name("valid_metric_name"), "valid_metric_name"
+        )
+        self.assertEqual(
+            sanitize_full_name("VALID_METRIC_NAME"), "VALID_METRIC_NAME"
+        )
+        self.assertEqual(
+            sanitize_full_name("_valid_metric_name"), "_valid_metric_name"
+        )
+        self.assertEqual(
+            sanitize_full_name("valid:metric_name"), "valid:metric_name"
+        )
+        self.assertEqual(
+            sanitize_full_name("valid_1_metric_name"), "valid_1_metric_name"
+        )
+        self.assertEqual(
+            sanitize_full_name("1leading_digit"), "_leading_digit"
+        )
+        self.assertEqual(
+            sanitize_full_name("1_~#consective_underscores"),
```

Comment:
There are no consecutive underscores here.
```suggestion
            sanitize_full_name("1_~#consective_invalid_characters"),
```

Reply:
We should have a test for consecutive underscores.

Reply:
This is making sure that a run on invalid characters only leads to one underscore instead of consecutive underscores.

I can add a case for just underscores.

Reply:
Added a test case.